src/utils/utils.js

```diff
@@ -400,12 +401,18 @@ export const getStartStopPatch = (vm, start) => {
 
 export const getResource = (
   model,
-  { name, namespaced = true, namespace, isList = true, matchLabels, matchExpressions } = {
+  { name, namespaced = true, namespace, isList = true, matchLabels, matchExpressions, prop } = {
     namespaced: true,
     isList: true,
   }
 ) => {
-  const res = { kind: model.kind, namespaced, namespace, isList, prop: model.kind };
+  const res = {
+    kind: model === NamespaceModel ? ProjectModel.kind : model.kind,
+    namespaced,
+    namespace,
+    isList,
+    prop: prop || model.kind,
```

Comment:
adding possibility to define own prop value which comes handy in case we have two queries for the same kind.

Reply:
is this going to work even with a ProjectModel?

Reply:
> wouldn't it be better to remove or at least limit the usage of getResource helper for now as discussed in [openshift/console#888](https://github.com/openshift/console/pull/888) ?

Yes, i guess we should but lets deal with it later. This PR is also going to be backported to 1.4 so its better to keep the changes minimal

Reply:
> is this going to work even with a ProjectModel?

Yes, it is. The way how `withResources` interacts with `firehose` requires to use the same `prop` as key in `resourceMap`. Example:
`templates: {
      resource: getResource(TemplateModel)
}`

then key `templates` is the same as `prop` from `getResources`

but if you have two queries for the same kind ..something like

`userTemplates: {
      resource: getResource(TemplateModel, {namespace: activeNamespace, prop: 'userTemplates', matchLabels: {[TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_VM}}),
    },
    commonTemplates: {
      resource: getResource(VmTemplateModel, {namespace: 'openshift', prop: 'commonTemplates', matchLabels: {[TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_BASE}}),
    }`

then you need to specify `prop` value otherwise both would end up with `templates` which would break stuff.

in case of ProjectModel, lets say you want to load namespaces
`namespaces: {
      resource: getResource(NamespaceModel),
}`
still works - key is `namespaces` and prop will be `namespaces`

if you want to load Projects
`projects: {
      resource: getResource(ProjectModel),
}`
again key is `projects` and prop will be `projects`

You can of course break it :
`projects: {
      resource: getResource(ProjectModel), {prop: 'somethingelse'},
}`
key is `projects` and prop will be `somethingelse` but i think we can live with that for now

Reply:
oki, makes sense